tajo-core/src/main/java/org/apache/tajo/engine/function/window/CurrentValue.java

```diff
@@ -24,41 +24,34 @@
 import org.apache.tajo.plan.function.WindowAggFunc;
 import org.apache.tajo.storage.Tuple;
 
-public abstract class FirstValue extends WindowAggFunc<Datum> {
+public abstract class CurrentValue extends WindowAggFunc<Datum> {
 
-  public FirstValue(Column[] columns) {
+  public CurrentValue(Column[] columns) {
     super(columns);
   }
 
   @Override
   public FunctionContext newContext() {
-    return new FirstValueContext();
+    return new CurrentValueContext();
   }
 
   @Override
   public void eval(FunctionContext ctx, Tuple params) {
-    FirstValueContext firstValueCtx = (FirstValueContext)ctx;
-    if(firstValueCtx.isSet == false) {
-      firstValueCtx.isSet = true;
-      if (params.get(0).isNotNull()) {
-        firstValueCtx.first = params.get(0);
+    CurrentValueContext currentValueCtx = (CurrentValueContext)ctx;
+      if (params.get(0) == null || params.get(0).isNotNull()) {
```

Comment:
Would you explain on this line?
I wonder why ```currentValueCtx .current``` becomes ```null``` instead of ```NullDatum``` when ```params.get(0)``` is null.